components/CalEPD/models/dke/depg750bn.cpp

```diff
@@ -0,0 +1,352 @@
+};
+
+
+// Constructor
+Depg750bn::Depg750bn(EpdSpi& dio):
+  Adafruit_GFX(DEPG750BN_WIDTH, DEPG750BN_HEIGHT),
+  Epd(DEPG750BN_WIDTH, DEPG750BN_HEIGHT), IO(dio)
+{
+  printf("Depg750bn() constructor injects IO and extends Adafruit_GFX(%d,%d)\n",
+  DEPG750BN_WIDTH, DEPG750BN_HEIGHT);
+}
+
+//Initialize the display
+void Depg750bn::init(bool debug)
+{
+    debug_enabled = debug;
+    if (debug_enabled) printf("Depg750bn::init(debug:%d)\n", debug);
+    //Initialize SPI at 4MHz frequency. true for debug
+    gpio_set_level((gpio_num_t)CONFIG_EINK_RST, ENABLE);
+
+    IO.init(4, debug);
+    fillScreen(EPD_WHITE);
+}
+
+void Depg750bn::fillScreen(uint16_t color)
+{
+  uint8_t data = (color == EPD_BLACK) ? DEPG750BN_8PIX_BLACK : DEPG750BN_8PIX_WHITE;
+
+  for (uint16_t x = 0; x < sizeof(_mono_buffer); x++)
+  {
+	  _mono_buffer[x] = data;
+  }
+}
+
+void Depg750bn::_wakeUp(){
+
+	IO.reset(10);
+	//  IO.cmd(0x12); // SWRESET
+	// Theoretically this display could be driven without RST pin connected
+	_waitBusy("SWRESET");
+
+	IO.cmd		(0x01); 					// POWER SETTING
+	IO.data 	(0x07);
+	IO.data 	(0x07); 					// VGH=20V,VGL=-20V
+	IO.data 	(0x3f); 					// VDH=15V
+	IO.data 	(0x3f); 					// VDL=-15V
+	IO.cmd		(0x04);
+	_waitBusy	("_wakeUp Power On");
+	IO.cmd		(0x00); 					//PANEL SETTING
+	IO.data		(0x1f); 					//KW: 3f, KWR: 2F, BWROTP: 0f, BWOTP: 1f
+	IO.cmd		(0x61); 					//tres
+	IO.data 	(DEPG750BN_WIDTH / 256); 	//source 800
+	IO.data 	(DEPG750BN_WIDTH % 256);
+	IO.data 	(DEPG750BN_HEIGHT / 256); 	//gate 480
+	IO.data 	(DEPG750BN_HEIGHT % 256);
+	IO.cmd		(0x15);
+	IO.data		(0x00);
+	IO.cmd		(0x50); 					//VCOM AND DATA INTERVAL SETTING
+	IO.data		(0x29);    					// LUTKW, N2OCP: copy new to old
+	IO.data		(0x07);
+	IO.cmd		(0x60); 					//TCON SETTING
+	IO.data		(0x22);
+	IO.cmd		(0x00); 					// panel setting
+	IO.data		(0x1f);    					// full update LUT from OTP
+
+}
+
+void Depg750bn::update()
+{
+  _using_partial_mode = false;
+
+  uint8_t xLineBytes = DEPG750BN_WIDTH/8;
+  uint8_t x1buf[xLineBytes];
+  _wakeUp();
+    // v2 SPI optimizing. Check: https://github.com/martinberlin/cale-idf/wiki/About-SPI-optimization
+  uint16_t i = 0;
+
+
+  IO.cmd(0x10);
+  printf("Sending a %d bytes buffer via SPI\n", (int)DEPG750BN_BUFFER_SIZE);
+  //sending data line by line
+  for(uint16_t y =  1; y <= DEPG750BN_HEIGHT; y++) {
+    for(uint16_t x = 1; x <= xLineBytes; x++) {
+      x1buf[x-1] = DEPG750BN_8PIX_WHITE;
+      if (x==xLineBytes) { // Flush the X line buffer to SPI
+        IO.data(x1buf,sizeof(x1buf));
+      }
+      ++i;
+    }
+  }
+  i = 0;
+  IO.cmd(0x13);
+  for(uint16_t y =  1; y <= DEPG750BN_HEIGHT; y++) {
+    for(uint16_t x = 1; x <= xLineBytes; x++) {
+      uint8_t data = _mono_buffer[i];
+      x1buf[x-1] = data;
+      if (x==xLineBytes) { // Flush the X line buffer to SPI
+        IO.data(x1buf,sizeof(x1buf));
+      }
+      ++i;
+    }
+  }
+  IO.cmd(0x12);
+  //Sending data byte by byte								//Clear display first
```

Comment:
You can just remove this commented area. Then to row morning I merge it.
Thanks and sorry for the last effort!

Reply:
I don't understand why should we remove 0x12 display refresh command after sending the frame data. is there a reason behind that ?

Reply:
Commented area has been removed.